### ISSUE TITLE
Order of imports/exports for a `trait` and a `trait as _` matters when it should not.

### ISSUE THREAD

**Issue by LorrensP-2158466** [rust-lang/rust#160742]
Found in #152822.

On ~~nightly~~ stable, this code will trigger the `ambiguously_glob_imported_trait` lint:
```rust
mod dsl {
    pub type IntoSql<T> = crate::expression::Foo<T>;
}

mod expression {
    pub struct Foo<T>(pub std::marker::PhantomData<T>);

    pub trait IntoSql {
        fn into_sql<T>(self) -> Foo<T>;
    }

    impl IntoSql for i32 {
        fn into_sql<T>(self) -> Foo<T> {
            Foo(std::marker::PhantomData)
        }
    }
}

mod prelude {
    pub use crate::expression::IntoSql;
    pub use crate::expression::IntoSql as _;
}

use dsl::*;
use prelude::*;

pub fn my_test_function() -> dsl::IntoSql<i32> {
    1i32.into_sql::<i32>()
}
```

But switching the order of imports/exports in `prelude` does not trigger the lint:
```rust
mod prelude {
    pub use crate::expression::IntoSql as _;
    pub use crate::expression::IntoSql;
}
```

I expected:

- the lint does not trigger in either cause
- the lint triggers in either case

cc @petrochenkov (since you reviewed both prs on the lint)
